src/compiler/parser.ts

```diff
@@ -666,6 +668,9 @@ namespace ts {
             sourceFile.languageVariant = getLanguageVariant(sourceFile.fileName);

```

Comment:
I think it is possible to have conflicting values of `languagevariant` and `scriptKind` since they might have different opinion whether file is `JSX` / `TSX` or not. - probably this also should check the value of `scriptKind` first and fallback to looking at file extension later